src/NexusMods.DataModel/Attributes/DiskState.cs

```diff
@@ -16,26 +17,25 @@ public static class DiskState
 {
     private static readonly string Namespace = "NexusMods.DataModel.DiskStateRegistry";
     
-    
     /// <summary>
     /// The associated game id.
     /// </summary>
-    public static readonly Attribute<GameDomain> Game = new($"{Namespace}/{nameof(Game)}", isIndexed: true, noHistory: true);
+    public static readonly GameDomainAttribute Game = new(Namespace, nameof(Game)) { IsIndexed = true, NoHistory = true };
     
     /// <summary>
     /// The game's root folder
     /// </summary>
-    public static readonly Attribute<AbsolutePath> Root = new($"{Namespace}/{nameof(Root)}", isIndexed: true, noHistory: true);
+    public static readonly StringAttribute Root = new(Namespace, nameof(Root)) { IsIndexed = true, NoHistory = true };
```

Comment:
This can't be an `AbsolutePath`?

Reply:
I think that is a LocationId

Reply:
I really should have documented this better, but the issue with using AbolutePath here is that AbsolutePath requires a IFileSystem, and these attributes don't have access to a DI Container at this location. Since they are static and part of the system as a whole, whenever we read the attribute as a string, we'll need some sort of IFileSystem to convert it to an AbsolutePath, and it's not clear here what the exact filesystem should be. 

Well LocationId requires a game and a version to link the data to, and in this context we don't have the version and the game is stored in a separate attribute. What the DiskStateRegistry really stores is: here's the last state (DiskStateTree) of this folder (AbsoltuePath), that was last associated with this Game and Loadout Revision. We could change this interface, but the previous version was coded to use `GameDomain+AbsolutePath` and I've simply broken these out into separate attributes

Reply:
Ah, yeah I read this wrong. Saving the absolute path is what we want since we need to identify the installation.
If it needs to be saved as a string oks